### PR TITLE
COMP: Address clang-tidy warnings in generated code

### PR DESCRIPTION
**Pull request by hjmjohnson**
Modernize the generateCLP main code, and generated code.

Updating the generated code allows programs using
SEM to build cleanly with clang-tidy.

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@jcfr   Do you have concerns with the change proposed?  These changes clear at least 6 warnings for every executable that uses SEM PARSE_ARGS at the begining of the main function.

**Comment by jcfr** [comment]
Thanks for the remainder and for working on this :pray: :100: 

Since the minimum version of C++ standard now supported is `11`, the proposed changes make sense.

https://github.com/Slicer/SlicerExecutionModel/blob/f19d6e88a94ba8f31ddafcff4adf185fe90d7e72/CMakeLists.txt#L19

File: ModuleDescriptionParser/Testing/TestData/parameter-file-with-pointfile-slicer-issue2979.params
Comment (jcfr) [inline ModuleDescriptionParser/Testing/TestData/parameter-file-with-pointfile-slicer-issue2979.params:2]: Nitpick: Could you check that this is only file with trailing spaces ? If not, could you make a dedicated commit fixing them ?

**Review by jcfr — CHANGES_REQUESTED** [review]
Let's address https://github.com/Slicer/SlicerExecutionModel/issues/136 before merging

File: ModuleDescriptionParser/Testing/TestData/parameter-file-with-pointfile-slicer-issue2979.params
Comment (hjmjohnson) [reply ModuleDescriptionParser/Testing/TestData/parameter-file-with-pointfile-slicer-issue2979.params:2]: Done.  Removed end of line spaces in all files (except tclap directory).

**Comment by hjmjohnson** [comment]
@jcfr  Can this now be merged?  It has been open without substantive changes or concerns since the last august.

**Comment by hjmjohnson** [comment]
@jcfr  PING.

**Comment by hjmjohnson** [comment]
@jcfr #136 was holding this up, but is now closed.  Can we please merge this PR now?

**Comment by hjmjohnson** [comment]
@jcfr @pieper Any chance that this can be rejected or accepted?  It has been passing for nearly a year.

**Comment by hjmjohnson** [comment]
ping.. silent little ping.. barely audible.

**Comment by jamesobutler** [comment]
@hjmjohnson Are there any additional changes you would like to make her considering Slicer defined the minimum to C++17 since you last pushed changes here. https://github.com/Slicer/Slicer/commit/dedb6c04d5f987d168e58dfda067717a7039b439

**Comment by jcfr** [comment]
Thanks for the reminder :pray: and patience.

Will integrate today :rocket:

**Pull request merged**
